cbcontainers/state/components/runtime_resolver_deployment.go

```diff
@@ -217,3 +232,27 @@ func (obj *ResolverDeploymentK8sObject) mutateVolumesMounts(container *coreV1.Co
 
 	commonState.MutateVolumeMountToIncludeRootCAsVolumeMount(container)
 }
+
+func (obj *ResolverDeploymentK8sObject) getDynamicReplicasCount(nodesToReplicasRatio *int32) (*int32, error) {
+	// Get the in-cluster config
+	config, err := rest.InClusterConfig()
```

Comment:
We already do the nodes list call on main.go
in order to get the k8s version
watch how we pass it to the function `NewStateApplier`
you pass another variable of NodesAmount or something (main.go:122)
and after that pass this value to the `NewResolverDeploymentK8sObject` function

Reply:
This needs to be called in the reconcile loop to get the latest updated nodes count.
I will propagate the client to `ResolverDeploymentK8sObject`